Common/Data/Custom/Intrinio/IntrinioEconomicData.cs

```diff
@@ -0,0 +1,228 @@
+        /// </summary>
+        /// <param name="dataTransformation">The item.</param>
+        public IntrinioEconomicData(IntrinioDataTransformation dataTransformation)
+        {
+            _dataTransformation = dataTransformation;
+        }
+
+
+        /// <summary>
+        ///     Return the URL string source of the file. This will be converted to a stream
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     String URL of source file.
+        /// </returns>
+        /// <remarks>
+        ///     Given Intrinio's API limits, we cannot make more than one CSV request per second. That's why in backtesting mode
+        ///     we make sure we make just one call to retrieve all the data needed. Also, to avoid the problem of many sources
+        ///     asking the data at the beginning of the algorithm, a pause of a second is added.
+        /// </remarks>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            SubscriptionDataSource subscription;
+            // We want to make just one call with all the data in backtesting mode.
+            if (_firstTime)
+            {
+                var order = string.Empty;
+                if (isLiveMode)
+                {
+                    // In Live mode, we only want the last observation.
+                    order = "desc";
+                }
+                else
+                {
+                    // In backtesting mode, we need the data in ascending order...
+                    order = "asc";
+                    // and avoid making furthers calls for the same data.
+                    _firstTime = false;
+                }
+
+                var item = GetStringForDataTransformation(_dataTransformation);
+                var url = string.Format("{0}identifier={1}&item={2}&sort_order={3}", _baseUrl,
+                                        config.Symbol.Value, item, order);
+                var byteKey = Encoding.ASCII.GetBytes(string.Format("{0}:{1}", _user, _password));
+                var authorizationHeaders = new List<KeyValuePair<string, string>>
+                {
+                    new KeyValuePair<string, string>("Authorization",
+                                                     string.Format("Basic ({0})", Convert.ToBase64String(byteKey)))
+                };
+
+                subscription = new SubscriptionDataSource(url, SubscriptionTransportMedium.RemoteFile, FileFormat.Csv,
+                                                          authorizationHeaders);
+                // Finally, here we force the engine to wait one second in case the we have multiples data sources added.
+                Thread.Sleep(millisecondsTimeout: 1000);
```

Comment:
//You can use a private class static to save the last API poll time and delay n-ms until at least 1s reached. 
// Doesn't scale well for 100 data sources though.. doesn't make Intrino very quant friendly.
public class IntrinioEconomicData : BaseData
{
   private static DateTime LastPollOfIntrinoAPI = new DateTime()
}